rama-http/src/headers/x_robots_tag_components/max_image_preview_setting.rs

```diff
@@ -0,0 +1,35 @@
+use rama_core::error::OpaqueError;
+use std::fmt::Formatter;
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub(super) enum MaxImagePreviewSetting {
+    None,
+    Standard,
+    Large,
+}
+
+impl std::fmt::Display for MaxImagePreviewSetting {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            MaxImagePreviewSetting::None => write!(f, "none"),
+            MaxImagePreviewSetting::Standard => write!(f, "standard"),
+            MaxImagePreviewSetting::Large => write!(f, "large"),
+        }
+    }
+}
+
+impl FromStr for MaxImagePreviewSetting {
+    type Err = OpaqueError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().trim() {
```

Comment:
This (`.to_lowercase()`) allocates a new string on the heap for nothing.
What I usually do is this:

```rust
let s = s.trim();

if s.eq_ignore_ascii_case("none") { ... } else if s.eq_ignore_ascii_case("..") { ... } ... else { Err(...) }
```

bit longer to type but no allocations required.